src/ui/wxWidgets/TreeCtrl.cpp

```diff
@@ -838,14 +839,50 @@ bool TreeCtrl::Remove(const CUUID &uuid)
 void TreeCtrl::SetItemImage(const wxTreeItemId &node,
                                const CItemData &item)
 {
-  // TODO: modify to display warning and expired states
+  int offset = 0;
+  CItemData::EntryType entrytype = item.GetEntryType();
+  
+  if((entrytype == CItemData::ET_NORMAL) || (entrytype == CItemData::ET_ALIASBASE) || (entrytype == CItemData::ET_SHORTCUTBASE)) {
+    time_t tttXTime(static_cast<time_t>(0));
```

Comment:
Minor: the method names GetXTime, GetPMTime, and GetCTime are not very intuitive and make the calculations here a bit hard to follow. Maybe we could at least name the corresponding local variables differently - e.g. `daysToExpiration` & `creationModificationTimestamp`?

Reply:
I added comments for those who are not so familiar with the short form of field names.